src/main/java/com/accelad/math/doubledouble/DoubleDouble.java

```diff
@@ -1221,8 +1220,12 @@ public int hashCode() {
     public boolean equals(Object object) {
         if (object instanceof DoubleDouble) {
             DoubleDouble that = (DoubleDouble) object;
-            return DoubleMath.fuzzyEquals(hi, that.hi, 1E-12) && DoubleMath.fuzzyEquals(lo, that.lo,
-                    1E-12);
+
+            if (that.hi == this.hi && that.lo == this.lo) return true;
+
+            DoubleDouble criteria = that.divide(DoubleDouble.fromOneDouble(1E29)).abs();
```

Comment:
Where does `1E29` come from? Could we give it a name ?

Reply:
Done